File: Practica1_i-Lunch/src/baseDatos/Serializador.java
package baseDatos;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.util.List;

import gestorAplicacion.gestionRestaurante.*;
import gestorAplicacion.usuariosRestaurante.*;

/* Se utiliza para serializar todos los objetos creados durante la ejecucion del proyecto
 * @author Erik Gonzalez
 * @author Felipe Miranda */

public class Serializador {
	/*
	 * Serializamos una lista por el nombre de la clase
	 * 
	 * @param <E> El generico se usa para poder agredar las clases que se crearon
	 * 
	 * @param lista Una lista de objetos
	 * 
	 * @param className El nombre de la clase que queremos usar como nombre del
	 * archivo
	 */

	public static <E> void serializar(List<E> lista, String className) {
		FileOutputStream fileOut;

		try {
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + className + ".txt";
			// Se crea un fileoutputstream para saber donde serializar los archivos
			fileOut = new FileOutputStream(path);
			// Se crea un objeto output stream para poder escribir en el archivo
			ObjectOutputStream out = new ObjectOutputStream(fileOut);
			// Guardamos la lista de objetos
			out.writeObject(lista);
			out.close();
			fileOut.close();

		} catch (FileNotFoundException e) {
			e.printStackTrace();

		} catch (IOException e) {
			e.printStackTrace();
		}
	}

	/* Serializamos todas las clases que necesitamos */
	public static void serializarTodo() {
		Serializador.serializar(Pedido.getPedidos(), "pedidos");
		Serializador.serializar(Producto.getProductos(), "productos");
		Serializador.serializar(Restaurante.getRestaurantes(), "restaurantes");
		Serializador.serializar(Cliente.getClientes(), "clientes");
		Serializador.serializar(Empleado.getEmpleados(), "empleados");
		Serializador.serializar(Administrador.getAdministradores(), "administradores");
		Serializador.serializar(Mesero.getMeseros(), "meseros");
		Serializador.serializar(Repartidor.getRepartidores(), "repartidores");
		Serializador.serializar(Chef.getChefs(), "chefs");
	}

}

File: Practica2_i-Lunch/src/baseDatos/serializador.py
import pathlib
import pickle
import os

from gestorAplicacion.usuariosRestaurante.empleado import Empleado
from gestorAplicacion.usuariosRestaurante.administrador import Administrador
from gestorAplicacion.usuariosRestaurante.repartidor import Repartidor
from gestorAplicacion.usuariosRestaurante.mesero import Mesero
from gestorAplicacion.usuariosRestaurante.chef import Chef
from gestorAplicacion.usuariosRestaurante.cliente import Cliente

from gestorAplicacion.gestionRestaurante.restaurante import Restaurante
from gestorAplicacion.gestionRestaurante.producto import Producto
from gestorAplicacion.gestionRestaurante.pedido import Pedido

# Codigo sacado del ejemplo proporcionado por el monitor

def serializar(lista, className):
    def camino(className):
        return os.path.join(pathlib.Path(__file__).parent.absolute(), f"temp\{className}.txt")

    try:
        # Crear el archivo pickle para guardar los objetos
        picklefile = open(camino(className), 'wb')

        # Pickle el objeto en el archivo
        pickle.dump(lista, picklefile)

        # Cerrar el archivo
        picklefile.close()
        
    except:
        print("Error de serializacion")

def serializarTodo():
    serializar(Empleado.getEmpleados(), "empleados")
    serializar(Administrador.getAdministradores(), "administradores")
    serializar(Repartidor.getRepartidores(), "repartidores")
    serializar(Mesero.getMeseros(), "meseros")
    serializar(Chef.getChefs(), "chefs")
    serializar(Cliente.getClientes(), "clientes")
    serializar(Restaurante.getRestaurantes(), "restaurantes")
    serializar(Producto.getProductos(), "productos")
    serializar(Pedido.getPedidos(), "pedidos")